File: pset1/mario/mario.c
#include <stdio.h>
#include <cs50.h>

void draw(int h);

int main(void)
{
    int height = get_int("Height: ");

    // if input is not between 1 and 8, re-prompt the user
    while (height < 1 || height > 8)
    {
        height = get_int("Height: ");
    }

    draw(height);
}

void draw(int h)
{
    for (int i = 1; i <= h; i++)
    {
        int spaceCount = h - i;

        // draw leading space
        for (int s = 1; s <= spaceCount; s++)
        {
            printf(" ");
        }
        // draw the first half of the bricks
        for (int c = 1; c <= i; c++)
        {
            printf("#");
        }
        // draw the gap
        printf("  ");
        // draw the second half of the bricks
        for (int c = 1; c <= i; c++)
        {
            printf("#");
        }
        // new line
        printf("\n");
    }
}

File: pset6/mario/more/mario.py
from cs50 import get_int


def main():
    height = get_height()
    draw_pyramid(height, height)


# Prompt user for the pyramid height
# between 1 and 8 inclusive
def get_height():
    while True:
        h = get_int('Height: ')
        if h >= 1 and h <= 8:
            break
    return h


# Draw pyramid
def draw_pyramid(row, h):
    if row == 0:
        return
    draw_pyramid(row - 1, h)
    # Draw leading spaces
    space_count = h - row
    print(' ' * space_count, end='')
    # Draw first half of bricks
    print('#' * row, end='')
    # Draw gap
    print('  ', end='')
    # Draw second half of bricks
    print('#' * row)


if __name__ == '__main__':
    main()